replication-api-impl/src/main/java/com/connexta/ion/replication/api/impl/data/SiteIndexImpl.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.ion.replication.api.impl.data;
+
+import com.connexta.ion.replication.api.data.SiteIndex;
+import java.util.Date;
+import java.util.UUID;
+import org.apache.commons.lang3.Validate;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.solr.core.mapping.Indexed;
+import org.springframework.data.solr.core.mapping.SolrDocument;
+
+/** Simple implementation of SiteIndex. */
+@SolrDocument(collection = "site_index")
+public class SiteIndexImpl implements SiteIndex {
```

Comment:
I didn't really see a need for it. No reason in particular though.

Reply:
It already provided support for the id and the version and later on, it will end up redefining the equivalent `fromMap()` and `toMap()` for conversion.

Reply:
Won't that equivalent functionality be in the managers?

Reply:
I extended it for now.